File: Examen_Corto/Octave/ejercicio6.m
% Solicitar al usuario que ingrese los lados del triángulo
lado1 = str2double(input("Ingrese el primer lado del triángulo: ", 's'));
lado2 = str2double(input("Ingrese el segundo lado del triángulo: ", 's'));
lado3 = str2double(input("Ingrese el tercer lado del triángulo: ", 's'));

% Inicializar la variable de salida
salida = "";

% Verificar si los lados forman un triángulo
if lado1 + lado2 > lado3 && lado1 + lado3 > lado2 && lado2 + lado3 > lado1
    % Determinar el tipo de triángulo
    if lado1 == lado2 && lado2 == lado3
        tipo_triangulo = "Equilátero";
    elseif lado1 == lado2 || lado1 == lado3 || lado2 == lado3
        tipo_triangulo = "Isósceles";
    else
        tipo_triangulo = "Escaleno";
    end

    % Concatenar el tipo de triángulo a la variable de salida
    salida = [salida, sprintf("El triángulo es de tipo: %s\n", tipo_triangulo)];

    % Almacenar el resultado en un archivo de texto
    fid = fopen('ejercicio6.txt', 'w');
    fprintf(fid, 'Tipo de triángulo= %s\n', tipo_triangulo);
    fprintf(fid, 'Lados del triángulo= %d, %d, %d\n', lado1, lado2, lado3);
    fclose(fid);

% Conexión a la base de datos PostgreSQL
pkg load database %cargar el paquete

conn = pq_connect(setdbopts('dbname','ExamenCorto1','host','localhost',
'port','5432','user','postgres','password','gabrielgrdb'))


query = sprintf("insert into salida values ('%s');",salida );
N = pq_exec_params(conn, query);



else
    salida = "Los lados proporcionados no forman un triángulo.";
end

% Mostrar la variable de salida
disp(salida);

File: Examen_Corto/Python/ejercicio6.py
# Solicitar al usuario que ingrese los lados del triángulo
lado1 = float(input("Ingrese el primer lado del triángulo: "))
lado2 = float(input("Ingrese el segundo lado del triángulo: "))
lado3 = float(input("Ingrese el tercer lado del triángulo: "))

# Inicializar la variable de salida
salida = ""

# Verificar si los lados forman un triángulo
if lado1 + lado2 > lado3 and lado1 + lado3 > lado2 and lado2 + lado3 > lado1:
    # Determinar el tipo de triángulo
    if lado1 == lado2 and lado2 == lado3:
        tipo_triangulo = "Equilátero"
    elif lado1 == lado2 or lado1 == lado3 or lado2 == lado3:
        tipo_triangulo = "Isósceles"
    else:
        tipo_triangulo = "Escaleno"

    # Concatenar el tipo de triángulo a la variable de salida
    salida += f"El triángulo es de tipo: {tipo_triangulo}\n"

    # Almacenar el resultado en un archivo de texto
    with open('ejercicio6.txt', 'w') as file:
        file.write(f'Tipo de triángulo= {tipo_triangulo}\n')
        file.write(f'Lados del triángulo= {lado1}, {lado2}, {lado3}\n')

    # Conexión a la base de datos PostgreSQL
    import psycopg2

    try:
        conn = psycopg2.connect(
            dbname='ExamenCorto1',
            host='localhost',
            port='5432',
            user='postgres',
            password='gabrielgrdb'
        )

        cursor = conn.cursor()

        query = f"INSERT INTO salida VALUES ('{salida}');"
        cursor.execute(query)

        conn.commit()

    except Exception as e:
        print(f"Error al conectar a la base de datos: {e}")

    finally:
        cursor.close()
        conn.close()

else:
    salida = "Los lados proporcionados no forman un triángulo."

# Mostrar la variable de salida
print(salida)
